_posts/2020-08-28-accesibility-in-mind.md

```diff
@@ -0,0 +1,74 @@
+
+For another example, consider guideline [2.4.7 - Focus Visible](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=132%2C211%2C311%2C325%2C131#focus-visible). It prescribes:
+
+*Any keyboard-operable user interface has a mode of operation where the keyboard focus indicator is visible.*
+
+Removing the keyboard-related parts, we are left with a piece of sound design advice fit for any situation:
+
+*Make the focus indicator visible.*
+
+Be it via touchscreen taps or mouse clicks, all users move from one part of an interface to the next. A visual cue to where they last interacted with helps them scan the surrounding elements with confidence, making the UI more effective. Such cues also help with switching to a whole different task and then back again, making the UI more usable.
+
+## 2. It improves maintainability
+
+Developers care about maintainability because they wish their designs to be ready for change. A tried and true practice for achieving maintainability is separation of concerns. In the world of front-end development, the main concerns to separate are content and presentation. On occasion, it is unclear where one ends and the other begins. It is there that WCAG helps draw the line. The most fundamental guideline [1.1 - Text Alternatives](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=132%2C211%2C311%2C325%2C131#text-alternatives) mandates:
+
+*All non-text content that is presented to the user has a text alternative that serves the equivalent purpose [...].*
+
+This forces the developer to ponder the content / presentation question. In the case of images, this translates to a decision on whether to use an  [`<img>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Img) tag or a [`background-image`](https://developer.mozilla.org/en-US/docs/Web/CSS/background-image) CSS attribute, making the value of the guideline all the more evident.
+
+## 3. It improves robustness
+
+Web apps today are called upon to achieve [Web Scale](https://en.wikipedia.org/wiki/Scalability). The preeminent architecture for achieving this is Microservices. Microservices however, require that developers adapt the way they put their apps together. The most radical change comes in handling failure states. The Microservices way of doing things is to expect that services might be unavailable and carry on nevertheless. This mindset change is aided by guideline [1.1 - Text Alternatives](https://www.w3.org/WAI/WCAG21/quickref/?showtechniques=132%2C211%2C311%2C325%2C131#text-alternatives) (see above). The UI of a modern Microservices-powered web app ought to gracefully fall back to a textual representation when fetching assets fails.
```

Comment:
I don't think it's necessary to capitalize "microservices".

Also, "microservice-powered".

Reply:
Done